apps/sft/main.py

```diff
@@ -111,25 +129,30 @@ def _init_dist(self):
 
     @endpoint
     async def setup(self):
-        self.train_dataloader = self.setup_data()
-        # self.train_dataloader = self.setup_data(
-        #     self.train_config.train_dataset_config,
-        #     self.train_config.train_dataloader_config,
-        #     self.train_config.packing_config,
-        # )
-        # self.val_dataloader = self.setup_data(
-        #     self.train_config.val_dataset_config,
-        #     self.train_config.val_dataloader_config,
-        #     self.train_config.packing_config,
-        # )
-
-        # TODO: confirm that this is working properly
-        # Should also use load, not dcp_load
+        # Setup training data from config
+        dataset_config = self.job_config.get("dataset")
+
+        self.train_dataloader = self.setup_data(
+            dataset_path=dataset_config.get("path"),
+            dataset_split=dataset_config.get("split"),
+        )
+
+        # Setup validation data from config
+        dataset_val_config = self.job_config.get("dataset_val", {})
+        self.val_dataloader = self.setup_data(
+            dataset_path=dataset_val_config.get("path", dataset_config.get("path")),
+            dataset_split=dataset_val_config.get("split", dataset_config.get("split")),
```

Comment:
nit but I don't like these nested .get calls. It also seems strange that we would fallback to validation on the training set. Personally I would just recommend checking if validation is enabled, and if it's not, don't even set up the validation dataloader at all.